### PR TITLE
bump 1.28

### PR DESCRIPTION
**Pull request by nfuden**
to 1.28

### CODE REVIEW

File: changelog/v1.28.1-patch1/bump-envoy.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+changelog:
+- type: DEPENDENCY_BUMP
+  dependencyOwner: solo-io
+  dependencyRepo: envoy-fork
+  dependencyTag: v1.28.1
+  resolvesIssue: false
+  description: >-
+    Update Envoy to latest from fork 1.28.1, 66ebbb61f317c01a6fa5d2c1be215a0213dbec84
```
Comment (jbohanon) [inline changelog/v1.28.1-patch1/bump-envoy.yaml:8]:
```suggestion
    Update Envoy to latest from forked 1.28.1
```

File: source/common/matcher/solo_matcher.cc [hunk 2]
```diff
@@ -38,8 +38,8 @@ class BaseMatcherImpl : public Matcher,
 
     matches &= Http::HeaderUtility::matchHeaders(headers, config_headers_);
     if (!config_query_parameters_.empty()) {
-      Http::Utility::QueryParams query_parameters =
-          Http::Utility::parseQueryString(
+      auto query_parameters =
```
Comment (ben-taussig-solo) [inline source/common/matcher/solo_matcher.cc:41]: is query_parameters no longer of type `Http::Utility::QueryParams`? Did the type name change?

File: ci/do_ci.sh [hunk 3]
```diff
@@ -19,7 +19,6 @@ cp -f "$UPSTREAM_ENVOY_SRCDIR/.bazelrc"                    "$SOURCE_DIR/upstream
 cp -f "$UPSTREAM_ENVOY_SRCDIR/.bazelversion"               "$SOURCE_DIR/.bazelversion"
 cp -f "$UPSTREAM_ENVOY_SRCDIR/ci/WORKSPACE.filter.example" "$SOURCE_DIR/ci/"
 cp -f "$UPSTREAM_ENVOY_SRCDIR/VERSION.txt"                 "$SOURCE_DIR/VERSION.txt"
-cp -f "$UPSTREAM_ENVOY_SRCDIR/tools/shell_utils.sh"        "$SOURCE_DIR/tools"
```
Comment (ben-taussig-solo) [inline ci/do_ci.sh:22]: Are we making this change because don't use this file? Or does it no longer exist at `$UPSTREAM_ENVOY_SRCDIR/tools/shell_utils.sh`?

File: source/extensions/filters/http/aws_lambda/config.cc [hunk 4]
```diff
@@ -126,7 +126,7 @@ void AWSLambdaConfigImpl::loadSTSData() {
         fmt::format("Web token file {} does not exist", token_file_));
   }
 
-  web_token_ = api_.fileSystem().fileReadToEnd(token_file_);
+  web_token_ = api_.fileSystem().fileReadToEnd(token_file_).value();
```
Comment (ben-taussig-solo) [inline source/extensions/filters/http/aws_lambda/config.cc:129]: Do we know why this change is necessary?

File: source/common/nats/streaming/BUILD [hunk 5]
```diff
@@ -49,6 +49,7 @@ envoy_cc_library(
     hdrs = ["connect_response_handler.h"],
     repository = "@envoy",
     deps = [
+        "@envoy//envoy/common:pure_lib",
```
Comment (ben-taussig-solo) [inline source/common/nats/streaming/BUILD:52]: Why is this change necessary? I can see that it was introduced with https://github.com/solo-io/envoy-gloo/commit/026e3cfdb3647d4d83c9a34fb5c7529d324c8c72 , but is this addition necessary and if so what does it do?

File: ci/do_ci.sh (hunk 3 above)
Comment (ben-taussig-solo) [reply ci/do_ci.sh:22]: The file no longer exists -- it was deleted upstream with https://github.com/envoyproxy/envoy/commit/07bf8f427386f4c95d658608f334f1f8c5a38a1c#diff-0671c871cea874cab5703e16d583fbb867d618fa82a92d0080fe07c92a692669

It looks like it used to contain python venv tooling that is no longer used/needed

File: source/common/nats/streaming/BUILD (hunk 5 above)
Comment (nfuden) [reply source/common/nats/streaming/BUILD:52]: Upstream sharded out some of its sub libraries so its no longer part of the base included stuff https://github.com/envoyproxy/envoy/commit/1a37802c9c8b2a163305774bbbf2ba21318531d2

File: bazel/repository_locations.bzl [hunk 6]
```diff
@@ -1,7 +1,7 @@
 REPOSITORY_LOCATIONS = dict(
     envoy = dict(
-        # envoy 1.27.3 from release v1.27.3-fork1
-        commit = "ebfb8ff1944e55756318f89bea189e0470daaed0",
+        # envoy 1.28.1 with backported ext_proc updates
+        commit = "a77b438ea9d7d3a2c23b77482d620f42f20a1562",
```
Comment (nfuden) [inline bazel/repository_locations.bzl:4]:
```suggestion
        commit = "47d4c36d399b9daae47cd7f6c4d41cf75e7e3ff8",
```

**Review by nfuden — COMMENTED** [review]
🚀

**Review by ben-taussig-solo — APPROVED** [review]
